### ISSUE TITLE
Latest Steam Beta refuses to download Proton

### ISSUE THREAD

**Issue by Niroc** [ValveSoftware/steam-for-linux#6045]
I'm not totally sure this is the correct place for this issue but, it directly affects Proton. 

If you change the global proton version or the game specific proton version, it will not attempt to download the specified proton. Even if you verify the games integrity, which used to work, or just turn steam proton off globally then turn it back on.

If you had multiple proton versions installed previously, swapping between them works just fine.

If you delete the proton folder found under ~/.local/share/Steam/steamapps/common/ then, it will not attempt to re-download proton if you use the game integrity checker or switch proton on and then back on within the steam settings.

The new feature of per-game "Proton version select by Valve testing" also, does not download proton. (In fact, this is how I encounter the issue.)

The only way I'm able to get Proton downloaded is to opt out of the Steam beta, get proton then, opt back in.

## System Information
- GPU: 870M
- Driver/LLVM version: Nvidia 415.25
- Kernel version: 4.19.14-1-MANJARO
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/Niroc/ab1f133f7e79e9e0d7932670b5f6cdc5

## Reproduction
Install the latest Steam beta (as of 20th Jan 2019) then try to install a Proton version not installed on your system.

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [  ] that I have checked whether there are updates for my system available.

**Comment by kisak-valve** [comment]
Hello @Niroc, are any other game updates or native game downloads affected?

Downloads and game updates are managed by the Steam client and not Proton, so I've transferred this issue report to the steam-for-linux issue tracker.

**Comment by Niroc** [comment]
Hi @kisak-valve, There was a couple of downloads between 100MB and 108MB in size for games that haven't had updates from their respective developer for a long time which was curious. Though, I used Dark Souls 3 for testing and it did not require one of these downloads. Also, the Proton download is around 350MB+ so, I don't believe it was related.

Thanks for transferring the issue to the correct place.

**Comment by jarrard** [comment]
I ended up with KCD being mostly deleted and Steam trying to download another 48GB... fortunately I had a backup on my HDD so I stopped it and was forced to reinstate the files.

Pretty sure KCD didn't get a update, its still on 1.7.2 last I checked. There were also small updates for other games.

**Comment by ensonic** [comment]
Same here. Proton 3.7.8 is selected, if I switch to 3.16.4 steam restarts but then the games don't launch and ~/.steam/steam/steamapps/common/ only has "Proton 3.7" and the open source project does not have 3.16.4 either (https://github.com/ValveSoftware/Proton/releases) - only 3.16.5 and .6.

**Comment by kisak-valve** [comment]
Hello @Niroc, @ensonic, can you read #6074 and let us know if it more accurately describes the issue you're seeing?

**Comment by ensonic** [comment]
@kisak-valve in my case settings offered 3.7.8 and 3.16.4, but only 3.7.8 was installed. After switching to 3.16.4 non of the games started. I assumed the steam client will automatically install 3.16.4 or only show installed versions. After I installed 3.16.4 manually everything is good. I have no idea if it is supposed to be auto-installed.

**Comment by jmakip** [comment]
For me games stopped launching after steam update. Steam had switched default from 3.7 to 3.16 but launch just flashed without running game, enabling proton debug logs didint produce any log file. Manually switching back to 3.7 for game titles fixed issue. After hour of searching found out that there is no proton 3.16 folder in steamapp/common but 3.7 exists. **Manually downloading Proton 3.16 from steam->library->tools fixes issue.**

**Comment by Niroc** [comment]
@kisak-valve Sorry for the delay. The issue does appear to be the same.
I set a native game (Cities Skylines) to run under steam proton then it downloads a copy of the Windows game files. However, I still can only use the versions of Proton previously installed on my system. if Proton X.YZ is not already installed then, it's not downloaded for use, the game just fails to launch. Even if the game integrity is check.

**Comment by jwatte** [comment]
This hit me with a Steam update version, not being in the Beta.
I had the same experience where games just stopped launching with no feedback.
Switching back to 3.6 also worked around the problem, because that version was actually downloaded.
Installing a Windows game AFTER I have selected a particular Proton version, downloads that version.

**Comment by parkerlreed** [comment]
Started completely fresh (wiped .steam* and .local/share/Steam)

Signed in, went to settings, and enabled proton for all games and left the latest beta selected. 

Restarted, downloaded Bejeweled 3. It pulled in Proton 3.7 into the common folder. Same for Steam Beta.

![image](https://user-images.githubusercontent.com/841440/53843392-38013100-3f70-11e9-8a0b-caf53b8cf554.png)

No matter what I do, the selected version in settings will not download. I've even tried setting a game to use that specifically and the game just doesn't run because the path doesn't exist (Proton beta not downloaded).

Arch Linux, steam 1.0.0.59-1

````
Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.0.0-arch1-1-ARCH
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12004000
    X Window Manager:  KWin
    Steam Runtime Version:  <Runtime disabled>
````

**Comment by kisak-valve** [comment]
Hello, per "Fixed more instances of 0-byte downloads and missing Proton or data files for Steam Play titles" in the [2019-03-06](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/1798531141142819058) Steam beta client update, please opt into Steam's beta if you're not already and retest this issue.

**Comment by parkerlreed** [comment]
Switched to 3.7 beta globally as a test and launched a Windows only title without any force versions. It downloaded correctly.
```
[parker@stealth ~]$ ls .local/share/Steam/steamapps/common/
'Bejeweled 3'  'Proton 3.16 Beta'  'Proton 3.7'  'Revenge of the Titans'   Steam.dll   vvvvvv
[parker@stealth ~]$ ls .local/share/Steam/steamapps/common/
'Bejeweled 3'       'Proton 3.7'       'Revenge of the Titans'   Voxelized
'Proton 3.16 Beta'  'Proton 3.7 Beta'   Steam.dll                vvvvvv
```

**Comment by unit73e** [comment]
Proton should probably always be installed as a tool.

All my Proton installations were in `$STEAM_APPS/common` but none showed as "not installed" in `Library => Tools`. With the recent update to Proton 4.2 not only did not any game update Proton but it was running my local Wine 4.4 installation instead for some odd reason. I fixed it by manually installing through `Library => Tools`. In fact I installed all versions from there.
